src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/TrackingHistoryOfAccessionsServiceTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.core;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.generators.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.AccessionStatus;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.BasicSpringDataRepositoryDatabaseService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.persistence.TestEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestRepositoryForAccessionsHistory;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestDatabaseServiceTestConfiguration.class})
+public class TrackingHistoryOfAccessionsServiceTest {
+
+    @Autowired
+    private TestRepositoryForAccessionsHistory repositoryForAccessionsHistory;
+
+    @Autowired
+    private BasicSpringDataRepositoryDatabaseService<TestModel, TestEntity, String, String> databaseService;
+
+    @Test
+    public void testAccessionCreations() throws AccessionCouldNotBeGeneratedException {
+
+        TrackingHistoryOfAccessionsService trackingHistoryOfAccessionsService = getTrackingDeprecationService();
+
+        trackingHistoryOfAccessionsService.createOrUpdateAccessions(Arrays.asList("Accession-1",
+                "Accession-2", "Accession-3"), AccessionStatus.CREATED, "");
```

Comment:
Personally I Would ban empty reasons, @cyenyxe any oppinion?

Reply:
I'm not sure, because it is going to be hard to find a reason for the `CREATED` status. Are we going to track **all** the creation events? That would at least double the size of the database.

This has already been a problem in EGA if I remember correctly.